opentelemetry-api/src/opentelemetry/util/_importlib_metadata.py

```diff
@@ -12,18 +12,92 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# FIXME: Use importlib.metadata when support for 3.11 is dropped if the rest of
-# the supported versions at that time have the same API.
-from importlib_metadata import (  # type: ignore
-    EntryPoint,
-    EntryPoints,
-    entry_points,
-    version,
-)
-
 # The importlib-metadata library has introduced breaking changes before to its
 # API, this module is kept just to act as a layer between the
 # importlib-metadata library and our project if in any case it is necessary to
 # do so.
 
-__all__ = ["entry_points", "version", "EntryPoint", "EntryPoints"]
+from sys import version_info
+
+if version_info < (3, 10):
```

Comment:
What is the benefit of not using `importlib_metadata` for newer Python versions?

I think this branch alone would solve all of our problems and you wouldn't need the custom `def entry_points()`.

Reply:
I might have missed something, but I see the main benefit as simply relying on a built-in module instead of external dependency, given that `importlib.metadata` is no longer provisional since Python 3.10 and we have a stable API. Also minimize issues related to the pinned version of `importlib-metadata` package

Reply:
I definitely see your point, Emedio, but my vote is for the simpler solution here until `importlib_metadata` is no longer needed. Seems like the cost is quite small.

Reply:
+1 on only relying on one library throughout the codebase. I am FOR using `importlib_metadata` for now for the reasons @aabmass  pointed out in terms of consistency. I am concerned that future contributions will not always use this library and use the standard built-in one instead, after which we must always remember as reviewers that we still support Python versions in which that will not be possible to do. This might be a calculated risk, as the contributions related to these libraries are relatively small, so if the vast majority of us agrees on that and the benefits of using a single, simple library out weigh the review/mental overhead of forcing users to use the backport then we can go ahead with this suggestion.

Keep in mind that if we do somehow want to go with the "custom built in entrypoint()" function of the current PR as of writing this review, our message to contributors in the future will be "use this custom thing we built", which technically is amounts to the same amount of headache/overhead as the previous problem.